src/main/java/org/openrewrite/jenkins/github/AddTeamToCodeowners.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.jenkins.github;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.MavenIsoVisitor;
+import org.openrewrite.text.PlainText;
+import org.openrewrite.text.PlainTextParser;
+import org.openrewrite.text.PlainTextVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class AddTeamToCodeowners extends ScanningRecipe<AddTeamToCodeowners.Scanned> {
+    private static final String FILE_PATH = ".github/CODEOWNERS";
+
+    @Override
+    public String getDisplayName() {
+        return "Add plugin developer team to CODEOWNERS";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds the `{artifactId}-plugin-developers` team to all files in `.github/CODEOWNERS` if absent.";
+    }
+
+    @Override
+    public Scanned getInitialValue(ExecutionContext ctx) {
+        return new Scanned(new ArtifactIdTeamNameGenerator());
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(Scanned acc) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(@Nullable Tree tree, ExecutionContext executionContext, Cursor parent) {
+                SourceFile sourceFile = (SourceFile) requireNonNull(tree);
```

Comment:
Not quite sure when tree would ever be null here; we can hope we don't find out. Did you see this pattern elsewhere?

Reply:
I believe so, though I don't recall where. I think it's mainly for telling the IDE to not highlight `tree` since that's marked `@Nullable` in the superclass.

Reply:
Found it in [CreateTextFile](https://github.com/openrewrite/rewrite/blob/main/rewrite-core/src/main/java/org/openrewrite/text/CreateTextFile.java#L73)